packages/react-native/ReactCommon/react/nativemodule/core/platform/android/ReactCommon/JavaTurboModule.cpp

```diff
@@ -807,18 +888,32 @@ jsi::Value JavaTurboModule::invokeJavaMethod(
               throw std::invalid_argument("Promise fn arg count must be 2");
             }
 
+            bool traceTurboModulePromiseRejectionEnabled = getFeatureFlagBoolValue("traceTurboModulePromiseRejectionEnabled");
+            std::shared_ptr<std::string> jsStack;
+            if (traceTurboModulePromiseRejectionEnabled) {
+                jsStack = std::make_shared<std::string>(createJSRuntimeError(runtime, "")
+                .asObject(runtime).getProperty(runtime, "stack").asString(runtime).utf8(runtime));
+            } else {
+                jsStack = std::make_shared<std::string>("");
+            }
+
             jsi::Function resolveJSIFn =
                 promiseConstructorArgs[0].getObject(runtime).getFunction(
                     runtime);
             jsi::Function rejectJSIFn =
                 promiseConstructorArgs[1].getObject(runtime).getFunction(
                     runtime);
+            jsi::Function rejectInternalJSIFn =
+                promiseConstructorArgs[1].getObject(runtime).getFunction(
+                    runtime);
+            auto rejectInternalWeakWrapper =
+                CallbackWrapper::createWeak(std::move(rejectInternalJSIFn), runtime, jsInvoker_);
```

Comment:
You can re-use the same jsi::Function ref (just make sure it's not moved elsewhere)

```suggestion
            jsi::Function rejectJSIFn =
                promiseConstructorArgs[1].getObject(runtime).getFunction(
                    runtime);
            auto rejectInternalWeakWrapper =
                CallbackWrapper::createWeak(rejectJSIFn, runtime, jsInvoker_);
```